compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -573,6 +573,8 @@ impl CodegenCx<'b, 'tcx> {
         ifn!("llvm.copysign.f64", fn(t_f64, t_f64) -> t_f64);
         ifn!("llvm.round.f32", fn(t_f32) -> t_f32);
         ifn!("llvm.round.f64", fn(t_f64) -> t_f64);
+        ifn!("llvm.roundeven.f32", fn(t_f32) -> t_f32);
+        ifn!("llvm.roundeven.f64", fn(t_f64) -> t_f64);
```

Comment:
@nagisa Can you confirm it's okay to add these llvm intrinsics?

Reply:
Yes. It is documented [here](https://llvm.org/docs/LangRef.html#llvm-roundeven-intrinsic). How available it is across different backends is a different question, however – a quick look suggests these lower to native instructions on aarch64 and x86 only (and maybe amdgpu). Everything else will lower to libm calls.